app/views/layouts/application.html.erb

```diff
@@ -11,6 +11,38 @@
   </head>
 
   <body>
+
+    <nav class="navbar navbar-expand-md navbar-dark bg-dark">
+      <div class="container-fluid">
+        <ul class="navbar-nav">
+           <li class="nav-item"><%= link_to "Create Event", new_event_path, class: "nav-link"%></li>
+        </ul>
+
+        <div class="mx-auto order-0">
+          <%= link_to "Private Events", root_path, class: "navbar-brand"%>
+        </div>
+
+       <div class="navbar-nav">
+         <ul class="navbar-nav ml-auto">
+           <%unless logged_in? %>
+             <li class="nav-item"><%= link_to "Login", login_path, class: "nav-link"%></li>
+             <li class="nav-item"><%= link_to "Sign Up", new_user_path, class: "nav-link" %></li>
+           <%end%>
+           <% if logged_in? %>
+             <li class="nav-item"><%= link_to user.username, user_path(session[:current_user_id]), class: "nav-link"%></li>
+             <li class="nav-item"><%= link_to "Log Out", logout_path, method: :delete, class: "nav-link"%></li>
+           <%end%>
```

Comment:
* [ ] There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. 
View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
**Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

Reply:
Removed login from views/application.html.erb by adding helper methods in application helpers to create a decision structure based on if the user is logged in or not.